SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.cc

```diff
@@ -174,14 +180,50 @@ Phase2TrackerDigitizerAlgorithm::SubdetEfficiencies::SubdetEfficiencies(const ed
   barrel_efficiencies = conf.getParameter<std::vector<double> >("EfficiencyFactors_Barrel");
   endcap_efficiencies = conf.getParameter<std::vector<double> >("EfficiencyFactors_Endcap");
 }
+void Phase2TrackerDigitizerAlgorithm::accumulateSimHits(std::vector<PSimHit>::const_iterator inputBegin,
+                                                        std::vector<PSimHit>::const_iterator inputEnd,
+                                                        const size_t inputBeginGlobalIndex,
+                                                        const uint32_t tofBin,
+                                                        const Phase2TrackerGeomDetUnit* pixdet,
+                                                        const GlobalVector& bfield) {
+  // produce SignalPoint's for all SimHit's in detector
+  // Loop over hits
+  uint32_t detId = pixdet->geographicalId().rawId();
+  size_t simHitGlobalIndex = inputBeginGlobalIndex;  // This needs to be stored to create the digi-sim link later
+
+  // find the relevant hits
+  std::vector<PSimHit> matchedSimHits;
+  std::copy_if(inputBegin, inputEnd, std::back_inserter(matchedSimHits), [detId](auto const& hit) -> bool {
+    return hit.detUnitId() == detId;
+  });
+  // loop over a much reduced set of SimHits
+  for (auto const& hit : matchedSimHits) {
+    LogDebug("Phase2DigitizerAlgorithm") << hit.particleType() << " " << hit.pabs() << " " << hit.energyLoss() << " "
+                                         << hit.tof() << " " << hit.trackId() << " " << hit.processType() << " "
+                                         << hit.detUnitId() << hit.entryPoint() << " " << hit.exitPoint();
+    double signalScale = 1.0;
+    // fill collection_points for this SimHit, indpendent of topology
+    if (select_hit(hit, (pixdet->surface().toGlobal(hit.localPosition()).mag() * c_inv), signalScale)) {
+      const auto ionization_points = primary_ionization(hit);  // fills ionization_points
```

Comment:
`const auto&` would be better (captures temporary object returned by the function and extends its lifetime to match the current scope, rather than making a copy)